File: backend/schemas/wrapped.py
from pydantic import BaseModel
from datetime import date
from typing import Optional, Any


class WordCloudWord(BaseModel):
    text: str
    value: int
    sentiment: Optional[str] = None


class WrappedTheme(BaseModel):
    theme: str
    count: int


class WrappedStats(BaseModel):
    entries_written: int
    positive_days: int
    top_habit_name: Optional[str] = None
    top_habit_streak: int
    avg_energy: str


class WrappedCorrelation(BaseModel):
    habit_name: str
    avg_sentiment_completed: float
    avg_sentiment_skipped: float
    correlation_delta: float


class WrappedReport(BaseModel):
    id: str
    period: str
    start_date: date
    end_date: date
    dominant_sentiment: Optional[str] = None
    dominant_sentiment_count: int
    total_days_journaled: int
    sentiment_timeline: list[dict]
    word_cloud_words: list[WordCloudWord]
    top_themes: list[WrappedTheme]
    habit_correlations: list[WrappedCorrelation]
    narrative: str
    stats: WrappedStats
    created_at: str


class GenerateWrappedRequest(BaseModel):
    period: str  # 'week' | 'month'
    start_date: date

File: frontend/src/types/wrapped.ts
export interface WordCloudWord {
  text: string
  value: number
  sentiment: string | null
}

export interface WrappedTheme {
  theme: string
  count: number
}

export interface WrappedStats {
  entries_written: number
  positive_days: number
  top_habit_name: string | null
  top_habit_streak: number
  avg_energy: string
}

export interface WrappedCorrelation {
  habit_name: string
  avg_sentiment_completed: number
  avg_sentiment_skipped: number
  correlation_delta: number
}

export interface WrappedReport {
  id: string
  period: string
  start_date: string
  end_date: string
  dominant_sentiment: string | null
  dominant_sentiment_count: number
  total_days_journaled: number
  sentiment_timeline: Array<{
    date: string
    primary_sentiment: string | null
    sentiment_score: number | null
    energy_level: string | null
    one_line_summary: string | null
  }>
  word_cloud_words: WordCloudWord[]
  top_themes: WrappedTheme[]
  habit_correlations: WrappedCorrelation[]
  narrative: string
  stats: WrappedStats
  created_at: string
}
